### PR TITLE
ccid: properly build with new meson build files

### LINKED ISSUE #349249: Yubikey not recognised by gpg
### Describe the bug
I keep getting the "insert smart-card with id ..." popup even though the smartcard is inserted when I try to sign a git commit.

### Steps To Reproduce
1. have a system on commit 4601c63145f2 or later with git signing enabled
2. try to make a commit
3. the popup keeps coming with "insert card with id" 

### Expected behavior
The inserted smartcard is recognised and can be used to sign the commit.

### Additional context
I've bisected the error to start on commit 4601c63145f2. I don't know what changed in ccid, but it broke somthing

### Notify maintainers
@anthonyroussel 

### Metadata

```console
> nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.3, NixOS, 24.11 (Vicuna), 24.11.20241016.4601c63`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/store/cz0kv05l66k9af2facij0gfk1zy3lnyy-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### LINKED ISSUE #350129: pkgs.ccid lacks proper files for pcscd usage
### Describe the bug

`pkgs.ccid` 1.6.1 is lacking these important files for pcscd usage:

* `etc/reader.conf.d/libccidtwin`
* `pcsc/drivers/serial/libccidtwin.so`

### Steps To Reproduce
1. `nix build 'nixpkgs#ccid'`
2. `tree -f ./result`

### Expected behavior

The files mentiond above is avaliable for usage in pcscd

A proper build of `pkgs.ccid` for reference:

```
@ nix build 'github:NixOS/nixpkgs/e08704965a52919348d9dba3570baf9c57befdf9#ccid'
@ tree -f ./result
./result
├── ./result/etc
│   └── ./result/etc/reader.conf.d
│       └── ./result/etc/reader.conf.d/libccidtwin
├── ./result/lib
│   └── ./result/lib/udev
│       └── ./result/lib/udev/rules.d
│           └── ./result/lib/udev/rules.d/92_pcscd_ccid.rules
└── ./result/pcsc
    └── ./result/pcsc/drivers
        ├── ./result/pcsc/drivers/ifd-ccid.bundle
        │   └── ./result/pcsc/drivers/ifd-ccid.bundle/Contents
        │       ├── ./result/pcsc/drivers/ifd-ccid.bundle/Contents/Info.plist
        │       └── ./result/pcsc/drivers/ifd-ccid.bundle/Contents/Linux
        │           └── ./result/pcsc/drivers/ifd-ccid.bundle/Contents/Linux/libccid.so
        └── ./result/pcsc/drivers/serial
            └── ./result/pcsc/drivers/serial/libccidtwin.so

12 directories, 5 files
```

### Additional context

I believe this is the base bug of #349249

People usually configures `scdaemon` in GnuPG with `disable-ccid`, per the manpage, it's required so GnuPG would use PC/SC driver (and thus pcscd) for smartcard communication. However with the files mentiond above missing, pcscd wouldn't be able to detect any smartcards thus identified no smartcards

### Notify maintainers

Cc @FliegendeWurst @anthonyroussel

### Metadata

```console
@ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.3-xanmod1, NixOS, 24.11 (Vicuna), 24.11.20241018.4c2fcb0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/riqkpszjqk02bi1wppfg8ip5xvh102qd-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by FliegendeWurst**
Fixes #350129
Fixes #349249 (probably)

Now builds all required files:
```
/nix/store/pqrl1al573qcpaspfbkbdrxlpgpd5qpk-ccid-1.6.1
├── etc
│   └── reader.conf.d
│       └── libccidtwin
├── lib
│   └── udev
│       └── rules.d
│           └── 92_pcscd_ccid.rules
└── pcsc
    └── drivers
        ├── ifd-ccid.bundle
        │   └── Contents
        │       ├── Info.plist
        │       └── Linux
        │           └── libccid.so
        └── serial
            └── libccidtwin.so
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/cc/ccid/package.nix [hunk 1]
```diff
@@ -23,13 +25,13 @@ stdenv.mkDerivation rec {
...
+    substituteInPlace meson.build --replace-fail \
+      "pcsc_dep.get_variable('usbdropdir')" \
+      "'$out/pcsc/drivers'"
   '';
 
-  configureFlags = [
-    "--enable-twinserial"
-    "--enable-serialconfdir=${placeholder "out"}/etc/reader.conf.d"
-    "--enable-ccidtwindir=${placeholder "out"}/pcsc/drivers/serial"
-    "--enable-usbdropdir=${placeholder "out"}/pcsc/drivers"
```
Comment (emilazy) [inline pkgs/by-name/cc/ccid/package.nix:32]: Do we know that these don’t matter any more? I see that the `usbdropdir` is handled, but what about the others?

File: pkgs/by-name/cc/ccid/package.nix [hunk 2]
```diff
@@ -23,13 +25,13 @@ stdenv.mkDerivation rec {
...
+  mesonFlags = [
+    "-Dserial=true"
```
Comment (emilazy) [inline pkgs/by-name/cc/ccid/package.nix:34]: Nit: `lib.mesonBool` is nice here.

File: pkgs/by-name/cc/ccid/package.nix [hunk 3]
```diff
@@ -23,13 +25,13 @@ stdenv.mkDerivation rec {
   postPatch = ''
     patchShebangs .
     substituteInPlace src/Makefile.am --replace-fail /bin/echo echo
```
Comment (emilazy) [inline pkgs/by-name/cc/ccid/package.nix:27]: Now dead code, presumably.

**Review by emilazy — COMMENTED** [review]
We should ideally have an `installCheckPhase` or a `passthru.tests.*` that would catch this problem if it happened again.

File: pkgs/by-name/cc/ccid/package.nix (hunk 1 above)
Comment (FliegendeWurst) [reply pkgs/by-name/cc/ccid/package.nix:32]: The first option is not needed as it is enabled by the meson flag below.
The other two are handled by the substitute.

**Comment by clerie** [comment]
I tried the change and can confirm that this fixes the issue with gpg smart cards not getting recognized.

**Comment by matdibu** [comment]
I can also confirm that this fixed it for me

**Review by anthonyroussel — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 350153`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ccid</li>
  </ul>
</details>

### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ccid</li>
  </ul>
</details>

**Comment by adamcstephens** [comment]
Thanks for fixing this!

**Comment by panchoh** [comment]
[Tracking](https://nixpk.gs/pr-tracker.html?pr=350153)

**Comment by korrat** [comment]
This seems to fix KeepassXC, but GPG is still broken.

**Pull request merged**